### PR TITLE
Add conceptual help build task for DSC resources

### LINKED ISSUE #122: Create a new build task for automatically create conceptual help
The function is here that first need to be moved to a better location so that a build task can use it: 
https://github.com/PowerShell/DscResource.Tests/blob/dev/DscResource.DocumentationHelper/PowerShellHelp.psm1

### PR DESCRIPTION
**Pull request by johlju**
# Pull Request

### Added
- Add conceptual build step for DSC resources (fixes #122).

## Task list

- [ ] The PR represents a single logical change. i.e. Cosmetic updates should go in different PRs.
- [ ] Added an entry under the Unreleased section of in the CHANGELOG.md as per [format](https://keepachangelog.com/en/1.0.0/).
- [ ] Local clean build passes without issue or fail tests (`build.ps1 -ResolveDependency`).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [ ] Unit tests added/updated. See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] Integration tests added/updated (where possible). See [DSC Resource Testing Guidelines](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md).
- [ ] New/changed code adheres to [DSC Resource Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md) and [Best Practices](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md).

---
This change is [Reviewable](https://reviewable.io/reviews/gaelcolas/sampler/124)

### CODE REVIEW

**Comment by johlju** [comment]
@gaelcolas That sounds like a plan. Is it as easy to move the task into a folder `tasks` in that module and Sampler (Invoke-Build) will pick it automatically or do I have to change anything in sampler to make that happen? Not so familiar with the build tasks yet. 🙂

**Comment by johlju** [comment]
Closing this as I have a working solution for the proposal that @gaelcolas made. Submitting a PR to DscResource.DocGenerator instead.

**Comment by gaelcolas** [comment]
Do generator needs to export an alias (like sampler does), and in Sampler you need to tell it to import it, as the templated build.yaml foes

**Comment by johlju** [comment]
Thanks! Yep, took be a couple of hours but got there. 😄 See PR on DocGenerator.

**Comment by johlju** [comment]
Ah, we should add this to the Plaster template of course. Re-opening this and will fix just the Plaster template according to the PR in DocGenerator.

**Comment by johlju** [comment]
I removed the build task that is being moved to DscResource.DocGenerator, so this PR now just updates the templates to use to new build task. 

This should not be merged until PR https://github.com/dsccommunity/DscResource.DocGenerator/pull/14 is merged.

**Comment by johlju** [comment]
This is ready for review now when the PR https://github.com/dsccommunity/DscResource.DocGenerator/pull/14 was merged and released.

**Comment by gaelcolas** [comment]
@johlju you can merge this one at your convenience.
Both PR looked fine.

File: Sampler/Assets/RequiredModules.psd1.template [hunk 1]
```diff
@@ -27,6 +27,8 @@
     'DscResource.Test'          = 'latest'
     'DscResource.AnalyzerRules' = 'latest'
     xDscResourceDesigner        = 'latest'
+    'DscResource.DocGenerator'  = 'latest'
```
Comment (gaelcolas) [inline Sampler/Assets/RequiredModules.psd1.template]: technically, this one should/could also be conditional of the Sampler module_type...

Comment (johlju) [reply Sampler/Assets/RequiredModules.psd1.template]: It is already, several of the modules above is within an if block for module type 'dsccommunity'. :)

**Pull request merged**